### ISSUE TITLE
Code 503 Back Again

### ISSUE THREAD

**Issue by shahhemal780** [PopcornTimeTV/PopcornTimeTV#856]
App version:
iOS/tvOS version:

Make sure these boxes are checked before submitting your issue - thanks!

- [ ] If app was installed using **Cocoapods**, make sure the latest version of master has been cloned (`git reset --hard; git pull`). If app was installed by downloading an **.ipa** file, please make sure you have the [latest version](https://github.com/PopcornTimeTV/PopcornTimeTV/releases/latest) installed.
- [ ] You have done a quick search for your issue already (detailed one is not necessary). If you are in doubt, post it anyway, it's not too big of a deal.
- [ ] You have read the [Issue Contributing Guidelines](https://github.com/PopcornTimeTV/PopcornTimeTV/blob/master/CONTRIBUTING.md#issues) for this repository.

#### Expected Behaviour

...

#### Actual Behaviour

...

#### Steps to reproduce the behaviour

1. ...
2. ...
3. ...

**Comment by 2bemeall1998Sept** [comment]
I'm have the same issue as of two days ago error 503 with popcorn time on apple 4k.

**Comment by 2bemeall1998Sept** [comment]
and those who are using popcorn time that I know are experiencing the same error code 503. could someone explain what's up?

**Comment by denkz** [comment]
Workaround to popcorntime. Not my work.


iOS
https://drive.google.com/file/d/1BVNoUPpXGf2NImtLgRDbctkcFDR2dgl-/view?usp=sharing

tvOS
https://drive.google.com/file/d/1loZcZjDnVD_BMa9qgRM6tbl6S8ZuUVI8/view?usp=sharing

**Comment by 2bemeall1998Sept** [comment]
Thank you.

**Comment by Bonovox78** [comment]
It’s fixed.

**Comment by dknightcc11** [comment]
YES! Thank you so much :)

**Comment by dknightcc11** [comment]
> Workaround to popcorntime. Done by FuzzyNutz
> 
> iOS
> https://drive.google.com/file/d/1BVNoUPpXGf2NImtLgRDbctkcFDR2dgl-/view?usp=sharing
> 
> tvOS
> https://drive.google.com/file/d/1loZcZjDnVD_BMa9qgRM6tbl6S8ZuUVI8/view?usp=sharing

Whats different about your build compared to the one online? Thank you for clarifying.

**Comment by rrikesh** [comment]
The normal app is working again now, no need to install the fork mentioned by @denkz

**Comment by 2bemeall1998Sept** [comment]
Cool, thanks for the update.

**Comment by Midifreakz** [comment]
> The normal app is working again now, no need to install the fork mentioned by @denkz

If i'm not mistaking, this fork utilizes the api over cloudflare, so the app keeps working when the server has downtime and you can still watch all the content that was available when the server went down. So i prefer having the FuzzyNutz version installed instead of the latest release. 

This functionality was also the latest commit made to the project by @angryDuck2

**Comment by drdoogong** [comment]
On tvOS the app is working however, it looks like it's still running a cached version. There haven't been any content updates since the end of November.

**Comment by Bonovox78** [comment]
Yeah it appears to be a cached version everywhere as it’s on my iPhone too and shows have not been updated ☹️

**Comment by Bonovox78** [comment]
Getting code 503 again. Anyone else?

**Comment by zojape** [comment]
again 503

**Comment by Midifreakz** [comment]
yes, down again.. The API has become really unstable these days..

**Comment by Dumeli** [comment]
@Midifreakz any way we could change the API to a more stable version ? Wish I new which one are used in other similar apps ..

**Comment by 2bemeall1998Sept** [comment]
Hi Everyone Popcorn is back up and running.

**Comment by 2bemeall1998Sept** [comment]
> and those who are using popcorn time that I know are experiencing the same error code 503. could someone explain what's up?

Hi everyone Popcorn Time is up again.

**Comment by Dumeli** [comment]
Down again .. guess I’ll stick to movie box pro ... what can be done to fix the 503 issue ? Can a different API be used like on all the other apps out there .. MovieBoxPro, Showbox or evening MediaBox ..??

**Comment by 2bemeall1998Sept** [comment]
Movie Box sounds like I would be interested in it, can I load it on my Apple TV?

**Comment by Dumeli** [comment]
> @Midifreakz any way we could change the API to a more stable version ? Wish I new which one are used in other similar apps ..



> Movie Box sounds like I would be interested in it, can I load it on my Apple TV?
> […](#)
https://www.movieboxpro.app/

**Comment by Dumeli** [comment]
@2bemeall1998Sept 

https://www.movieboxpro.app/

**Comment by Midifreakz** [comment]
> @Midifreakz any way we could change the API to a more stable version ? Wish I new which one are used in other similar apps ..

_I must first state that I am not a developer whatsoever but just an enthousiast. I have gathered some knowledge just by spending hours of reading on Github and numerous fora._

So as I believe things are right now: 

Popcorn time uses an API that scrapes torrents from certain torrent websites and is entirely community driven. As far as I know, there is only one working API hosted at this time. I have no idea if there is an individual responsible for hosting this API or if it is a group thing. What I do know: All versions of popcorn time are using this exact same API for the moment.

The reason why some versions of PT, such as the desktop versions are 'always online', even when the latest release of PopcorntimeTV is giving a 503 error is because these other PT versions are not using a direct link to that API but rather a link to a Cloudflare backup of this API. 

A Cloudflare backup will show a version of the API with everything more or less working like if the 'master' api was working.

When the actual API comes online again, Cloudflare updates the backup to the most recent state.

_In other words:_ as long as the api comes up again from time to time, having a version of PT with a cloudflare backup setup will always look like its working. unless the API stays offline for way to long. Then Cloudflare will probably flush the stored cached version. 

For the moment it seems like the development has stalled on this fork of PopcorntimeTV.
However, a developer called rimonhanna created another fork and made a version of popcorn time including the Cloudflare fallback. He and other people also made a pull request to integrate this functionality to this fork of PT and it was the last commit made by a developer on this fork. dating from a few months ago. So when you build your own version of this PT fork, you will have the Cloudflare fallback. 

FIY: Here's a list with all the forks of PT: 
https://techgaun.github.io/active-forks/index.html#PopcornTimeTV/PopcornTimeTV

Services like MovieBoxPro do not use an API that would be compatible with PopcornTime.
I have tested MovieboxPro and it looks like this is managed in an entirely different way. They include Cam rips and update the torrents as BluRay or WEB versions come out. 

Be warned: They seem like a profit company and not a community based app. This means that some techsavvi individuals are having your information. If they get shut down they will probably be forced to hand over their user database to authorities as has happened with other platforms before.


I am not stating here that anything I said is absolute truth. If anyone with more devskills has a more definitive answer on any of this, please enlighten us! ;)

**Comment by Dumeli** [comment]
> > @Midifreakz any way we could change the API to a more stable version ? Wish I new which one are used in other similar apps ..
> 
> _I must first state that I am not a developer whatsoever but just an enthousiast. I have gathered some knowledge just by spending hours of reading on Github and numerous fora._
> 
> So as I believe things are right now:
> 
> Popcorn time uses an API that scrapes torrents from certain torrent websites and is entirely community driven. As far as I know, there is only one working API hosted at this time. I have no idea if there is an individual responsible for hosting this API or if it is a group thing. What I do know: All versions of popcorn time are using this exact same API for the moment.
> 
> The reason why some versions of PT, such as the desktop versions are 'always online', even when the latest release of PopcorntimeTV is giving a 503 error is because these other PT versions are not using a direct link to that API but rather a link to a Cloudflare backup of this API.
> 
> A Cloudflare backup will show a version of the API with everything more or less working like if the 'master' api was working.
> 
> When the actual API comes online again, Cloudflare updates the backup to the most recent state.
> 
> _In other words:_ as long as the api comes up again from time to time, having a version of PT with a cloudflare backup setup will always look like its working. unless the API stays offline for way to long. Then Cloudflare will probably flush the stored cached version.
> 
> For the moment it seems like the development has stalled on this fork of PopcorntimeTV.
> However, a developer called rimonhanna created another fork and made a version of popcorn time including the Cloudflare fallback. He and other people also made a pull request to integrate this functionality to this fork of PT and it was the last commit made by a developer on this fork. dating from a few months ago. So when you build your own version of this PT fork, you will have the Cloudflare fallback.
> 
> FIY: Here's a list with all the forks of PT:
> https://techgaun.github.io/active-forks/index.html#PopcornTimeTV/PopcornTimeTV
> 
> Services like MovieBoxPro do not use an API that would be compatible with PopcornTime.
> I have tested MovieboxPro and it looks like this is managed in an entirely different way. They include Cam rips and update the torrents as BluRay or WEB versions come out.
> 
> I also think they are hosting these video files somewhere as loading times are insanely fast... It all seems a bit too good to be true and I am affraid it will get shut down sooner or later.
> 
> Also, be warned: They seem like a profit company and not a community based app. This means that some techsavvi individuals are having your information. If they get shut down they will probably be forced to hand over their user database to authorities as has happened with other platforms before.
> 
> I am not stating here that anything I said is absolute truth. If anyone with more skills and knowledge has a more definitive answer on any of this, please enlighten us! ;)

Great text much appreciated

**Comment by Bonovox78** [comment]
Gone again 😩

**Comment by 2bemeall1998Sept** [comment]
Yes error 503 again

**Comment by Bonovox78** [comment]
It’s back again!

**Comment by 2bemeall1998Sept** [comment]
Yes it is.

**Comment by Bonovox78** [comment]
Fixed again 😀

**Comment by 2bemeall1998Sept** [comment]
Excellent 😊👍👍 two thumbs up.

**Comment by Bonovox78** [comment]
Not that it’s big deal but do none of the trailers work for any movie?

**Comment by 2bemeall1998Sept** [comment]
Some of them do.

**Comment by Bonovox78** [comment]
> Some of them do.
> […](#)

I tried about 13 movies from the main page and no trailers are playing.

**Comment by 2bemeall1998Sept** [comment]
Error 503 once again.

**Comment by Bonovox78** [comment]
Same here 😢

**Comment by 2bemeall1998Sept** [comment]
Popcorn time is back up 😀👍

**Comment by Bonovox78** [comment]
503 unknown error is back.

**Comment by Bonovox78** [comment]
It’s fixed again.

**Comment by Dumeli** [comment]
> It’s fixed again.

👍

**Comment by 2bemeall1998Sept** [comment]
Yes it’s Back, thank you.

**Comment by 2bemeall1998Sept** [comment]
Hello is Popcorn Time down again on Apple TV’s or is it down for everyone?
